lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/IFileDeSerializer.cs

```diff
@@ -30,6 +34,17 @@ public interface IFileDeSerializer<T>
         /// </summary>
         /// <param name="fileFolder"></param>
         /// <returns></returns>
+        [Obsolete("Remove after 0.14")]
         T Deserialize(string fileFolder);
+
+        /// <summary>
+        /// The input is a set of file paths. 
+        /// The output is of type T which is defined by the client.
+        /// If there is any IO error, IOException could be thrown.
+        /// </summary>
+        /// <param name="filePaths"></param>
+        /// <param name="local"></param>
+        /// <returns></returns>
+        T Deserialize(ISet<string> filePaths, bool local);
```

Comment:
It is a named parameter from FileSystemInputPartition, then passed to the deserializer. 

Most cases, deserializer only needs one way to get files, wither from remote or copied local files. But if we have a flag, we can easily switch between the two in case we need. We can even  compare the performance between the two.